crates/hyperdrive-math/src/long/fees.rs

```diff
@@ -46,12 +51,33 @@ impl State {
     pub fn close_long_flat_fee(
         &self,
         bond_amount: FixedPoint,
-        normalized_time_remaining: FixedPoint,
+        maturity_time: U256,
+        current_time:U256 
     ) -> FixedPoint {
+        let normalized_time_remaining = self.calculate_time_remaining(maturity_time, current_time);
         // flat_fee = (d_y * (1 - t) * phi_flat) / c
         bond_amount.mul_div_down(
             fixed!(1e18) - normalized_time_remaining,
             self.vault_share_price(),
         ) * self.flat_fee()
     }
+
+    fn calculate_time_remaining(
+        &self,
+        maturity_time: U256,
+        current_time: U256 
+    ) -> FixedPoint {
+        let latest_checkpoint = self.to_checkpoint(current_time); 
+        let time_remaining = if maturity_time > latest_checkpoint {
+            fixed!(maturity_time - latest_checkpoint)
```

Comment:
Wasn't sure how to convert this to FixedPoint.

<img width="797" alt="image" src="https://github.com/delvtech/hyperdrive/assets/4524175/d6da844e-52f6-42aa-8561-909f11c615ff">

Reply:
Fixed, h/t @jalextowle